src/main/java/nextstep/subway/domain/line/Line.java

```diff
@@ -0,0 +1,240 @@
+    private String color;
+    @OneToMany(cascade = CascadeType.ALL)
+    @JoinColumn(name = "line_id")
+    private List<Section> sections = new ArrayList<>();
+
+    public Line() {
+    }
+
+    public Line(String name, String color, Section section) {
+        this.name = name;
+        this.color = color;
+        sections.add(section);
+    }
+
+    public void update(String name, String color) {
+        this.name = name;
+        this.color = color;
+    }
+
+    public void addSection(Section newSection) {
+        // 구간과 구간 내의 역이 노선에 이미 등록되어 있는 경우를 검증
+        validateNewSection(newSection);
+
+        if (isFirstAddCase(newSection)) {
+            addSectionToFirst(newSection);
+            return;
+        }
+
+        if (isLastAddCase(newSection)) {
+            addSectionToLast(newSection);
+            return;
+        }
+
+        if (isMiddleAddCase(newSection)) {
+            addSectionToMiddle(newSection);
+            return;
+        }
+
+        throw new IllegalArgumentException("추가할 구간의 역이 유효하지 않습니다.");
+    }
+
+    private void validateNewSection(Section newSection) {
+        Station upStation = newSection.getUpwardStation();
+        Station downStation = newSection.getDownwardStation();
+
+        List<Station> stations = sections.stream()
+                .flatMap(section -> Stream.of(section.getUpwardStation(), section.getDownwardStation()))
+                .distinct()
+                .collect(Collectors.toList());
+
+        if (stations.contains(upStation) && stations.contains(downStation)) {
+            throw new IllegalArgumentException("추가할 구간의 역이 모두 노선에 등록되어 있습니다.");
+        }
+
+        if (sections.contains(newSection)) {
+            throw new IllegalArgumentException("추가할 구간이 이미 노선에 등록되어 있습니다.");
+        }
+    }
+
+    private boolean isFirstAddCase(Section newSection) {
+        int sectionsSize = sections.size();
+
+        if (sectionsSize == 0) {
+            throw new IllegalArgumentException("노선에 존재하는 구간이 없습니다.");
+        }
+
+        Section firstSection = sections.get(0);
+        return firstSection.getUpwardStation().equals(newSection.getDownwardStation());
+    }
+
+    private boolean isLastAddCase(Section newSection) {
+        int sectionsSize = sections.size();
+
+        if (sectionsSize == 0) {
+            throw new IllegalArgumentException("노선에 존재하는 구간이 없습니다.");
+        }
+
+        Section lastSection = sections.get(sectionsSize - 1);
+        return lastSection.getDownwardStation().equals(newSection.getUpwardStation());
+    }
+
+    private boolean isMiddleAddCase(Section newSection) {
+        if (sections.stream()
+                .anyMatch(e -> e.getUpwardStation().equals(newSection.getUpwardStation()))) {
+            return true;
+        }
+
+        if (sections.stream()
+                .anyMatch(e -> e.getDownwardStation().equals(newSection.getDownwardStation()))) {
+            return true;
+        }
+
+        return false;
```

Comment:
아래와 같이 축약해서 작성해볼 수 있을 것 같아요.
```suggestion
        return sections.stream()
                .anyMatch(e -> e.getDownwardStation().equals(newSection.getDownwardStation()));
```